pkg/harvester/components/settings/rancher-cluster.vue

```diff
@@ -0,0 +1,265 @@
+
+      this.existingSecret = secrets.find((secret) => secret.metadata.name === 'rancher-cluster-config' &&
+        secret.metadata.namespace === 'harvester-system'
+      );
+      if (this.existingSecret && this.existingSecret.data && this.existingSecret.data.kubeConfig) {
+        const decodedContent = atob(this.existingSecret.data.kubeConfig);
+
+        this.parseDefaultValue.kubeConfig = decodedContent;
+        this.$nextTick(() => {
+          this.update();
+        });
+      }
+    },
+
+    async createOrUpdateRancherKubeConfigSecret() {
+      this.errors = [];
+      this.isCreatingSecret = true;
+      if (!this.parseDefaultValue.kubeConfig) {
+        this.errors.push(this.t('validation.required', { key: this.t('harvester.setting.rancherCluster.kubeConfig') }, true));
+        this.isCreatingSecret = false;
+
+        return Promise.reject(this.errors);
+      }
+      try {
+        const inStore = this.$store.getters['currentProduct'].inStore;
+        let secret;
+        const isUpdate = !!this.existingSecret;
+
+        if (this.existingSecret) {
+          secret = this.existingSecret;
+          secret.setData('kubeConfig', this.parseDefaultValue.kubeConfig);
+        } else {
+          secret = await this.$store.dispatch(`${ inStore }/create`, {
+            apiVersion: 'v1',
+            kind:       'Secret',
+            metadata:   {
+              name:      'rancher-cluster-config',
+              namespace: 'harvester-system'
+            },
+            type: 'secret',
+            data: { kubeConfig: btoa(this.parseDefaultValue.kubeConfig) }
+          });
+        }
+        await secret.save();
+        await this.checkExistingSecret();
+        this.$store.dispatch('growl/success', { message: isUpdate ? this.t('harvester.setting.rancherCluster.secretUpdated') : this.t('harvester.setting.rancherCluster.secretCreated') }, { root: true });
+        this.isCreatingSecret = false;
+
+        return Promise.resolve();
+      } catch (err) {
+        this.errors = exceptionToErrorsArray(err);
+        this.isCreatingSecret = false;
+
+        return Promise.reject(this.errors);
+      }
+    },
+
+    async willSave() {
+      // Only send secret API if enabled
+      if (this.parseDefaultValue.removeUpstreamClusterWhenNamespaceIsDeleted) {
+        await this.createOrUpdateRancherKubeConfigSecret();
+      }
+      // If disabled, do nothing for secret
+      this.update();
+
+      return Promise.resolve();
+    },
+
+    useDefault() {
+      let defaultVal = false;
+
+      try {
+        const parsed = JSON.parse(this.value.default || '{}');
+
+        defaultVal = parsed.removeUpstreamClusterWhenNamespaceIsDeleted || false;
+      } catch (e) {
+        defaultVal = false;
+      }
+      this.parseDefaultValue.removeUpstreamClusterWhenNamespaceIsDeleted = defaultVal;
+      if (!defaultVal) {
+        this.parseDefaultValue.kubeConfig = '';
+      }
+      this.update();
+    }
```

Comment:
When clicking `Use the default value`, shouldn't we reset it to `{}`? because the default value from `harvesterhci.io.settings` is `{}`.